### ISSUE TITLE
ScriptSharp 0.7.1 and AjaxRequestCallback<object> compilation fails

### ISSUE THREAD

**Issue by stevehobbsdev** [nikhilk/scriptsharp#32]
I'm attempting to upgrade my version of ScriptSharp from 0.6.2 to 0.7.1 to take advantage of the new generics feature. The last issue I have come across that I don't know how to solve is this: when making an Ajax call, the compiler fails with the message "Check that your C# source compiles and that you are not using an unsupported feature".

Here's the code:

```
jQuery.Ajax(
        new jQueryAjaxOptions(
            "url", url,
            "type", "GET",
            "dataType", "html",
            "contentType", "application/json",
            "processData", false,
            "success", (AjaxRequestCallback<object>)delegate(object result, string textData, jQueryXmlHttpRequest<object> XMLHttpRequest)
            {
                container.Html(XMLHttpRequest.ResponseText);

                Type.InvokeMethod(null, "ajaxUnBlock", null);
            },
            "error", (AjaxErrorCallback<object>)delegate(jQueryXmlHttpRequest<object> request, string textData, Exception e)
            {
                //Debug.TraceDump(XMLHttpRequest);
            }
        )
    );
```

I know it's the delegates which are causing this. If I was to instead populate the properties of jQueryAjaxOptions (as opposed to providing a dictionary of key/values) then when I double-click the error in the error window it points me to the line which specifies the success delegate.

I seem to be able to work around it by creating a method instead of using an anonymous delegate and the compiler error goes away, but I would like to avoid going down that road if at all possible as I have a large amount of code which would have to be converted in this way.

OR... Making the function which contains this anonymous delegate static fixes the issue. While researching this issue I found a closed issue on Github (https://github.com/NikhilK/scriptsharp/issues/22) which seems related - the only thing I'm unsure of now is whether this behaviour is intended or not.

**Comment by nikhilk** [comment]
Unfortunately the limited generics implementation is starting to get in the way.

I am going to define non-generic versions of AjaxRequestCallback, and other delegates, which implies, we'll also need non-generic version of jQueryXmlHttpRequest. This should be fine for jQueryAjaxOptions, since it was using "object" for the TData generic parameter in the callback and jQueryXmlHttpRequest anyway.

**Comment by stevehobbsdev** [comment]
Thanks for this. I've since downloaded the latest source, recompiled and copied the new dll's over my installation and now (after having removed the generic implementations) everything works.

**Comment by nikhilk** [comment]
Fix included in 0.7.2.0 release.
